### PR TITLE
Core api multiple entries error

### PR DESCRIPTION
**Pull request by anas-yousef**
## Status
- [x] In Progress
- [ ] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes: https://jira-dc.paloaltonetworks.com/browse/XSUP-31868

## Description
Added the option to run the command **core-api-multipart** subsequently in the playbook **Process Email - Generic v2** by using the automation **UploadFile** with the option **isArray: true**.
## Must have
- [ ] Tests
- [ ] Documentation

### CODE REVIEW

File: Packs/DemistoRESTAPI/Scripts/UploadFile/UploadFile.py [hunk 1]
```diff
@@ -8,30 +8,31 @@ def upload_file(incident_id: str, entry_id: str, body: str = '', as_incident_att
...
-    if uploaded_entry_id:
-        readable += f' Entry ID is {uploaded_entry_id}'
-    if body:
-        readable += f'. Comment is:{body}'
-
-    return readable, response
+    for entry_id in entry_ids:
+        response = upload_file(incident_id, entry_id, body, target == 'incident attachment')
+        if is_error(response[0]):
+            raise Exception("There was an issue uploading the file. Check your API key and input arguments.")
```
Comment (JasBeilin) [inline Packs/DemistoRESTAPI/Scripts/UploadFile/UploadFile.py:20]:
```suggestion
            raise DemistoException("There was an issue uploading the file. Check your API key and input arguments.")
```

File: Packs/DemistoRESTAPI/Scripts/UploadFile/UploadFile_test.py
Comment (JasBeilin) [inline Packs/DemistoRESTAPI/Scripts/UploadFile/UploadFile_test.py:1]: Lets also have test for one file as we just changed the behavior.

File: Packs/Phishing/Playbooks/Process_Email_-_Generic_v2.yml [hunk 2]
```diff
@@ -2789,10 +2789,11 @@ tasks:
       id: 754742b3-8a98-45b3-8d02-a850f90d83fd
       version: -1
       name: Upload email attachments to layout
-      description: Uploads file attachments of the email to the incident layout.
-      script: '|||core-api-multipart'
+      description: Copies a file from this incident to the specified incident. The
+        file is recorded as an entry in the specified incident’s War Room.
+      scriptName: UploadFile
```
Comment (JasBeilin) [inline Packs/Phishing/Playbooks/Process_Email_-_Generic_v2.yml:2794]: Was this designed with the security team?

File: Packs/Phishing/Playbooks/Process_Email_-_Generic_v2.yml [hunk 3]
```diff
@@ -2789,10 +2789,11 @@ tasks:
...
       type: regular
-      iscommand: true
+      iscommand: false
```
Comment (JasBeilin) [inline Packs/Phishing/Playbooks/Process_Email_-_Generic_v2.yml:2796]: Is an automation does not qualify as command in the PB anymore?

**Review by JasBeilin — CHANGES_REQUESTED** [review]
Looks good, please see my comments.

File: Packs/Phishing/Playbooks/Process_Email_-_Generic_v2.yml (hunk 3 above)
Comment (anas-yousef) [reply Packs/Phishing/Playbooks/Process_Email_-_Generic_v2.yml:2796]: [Example](https://github.com/demisto/content/blob/master/Packs/DeveloperTools/TestPlaybooks/playbook-EDL_Performance_Test_-_Concurrency.yml#L436), I believe if the task has an automation, the **iscommand** should be false.

**Review by idovandijk — COMMENTED** [review]
Changes look OK, but do we still need to check if the Core API integration is enabled if we use this script?
Also was this tested successfully on XSOAR 8? You can forward an EML file to a phishing inbox, it should trigger that flow in the playbook

**Comment by anas-yousef** [comment]
I believe so, since the script [UploadFile](https://github.com/demisto/content/blob/master/Packs/DemistoRESTAPI/Scripts/UploadFile/UploadFile.py#L7) relies on the command **core-api-multipart**.

**Comment by anas-yousef** [comment]
Concerning testing this on XSOAR 8, I checked the script **UploadFile** by creating a new playbook that uses it, and saw that it is able to run on multiple entry IDs, and upload the attachments to the relevant incident.
@idovandijk I sent you in private so we can test the whole flow of the playbook in XSOAR 8

**Comment by idovandijk** [comment]
> Concerning testing this on XSOAR 8, I checked the script **UploadFile** by creating a new playbook that uses it, and saw that it is able to run on multiple entry IDs, and upload the attachments to the relevant incident. @idovandijk I sent you in private so we can test the whole flow of the playbook in XSOAR 8

Need to verify that the newly uploaded files show up in the layout

File: Packs/Phishing/Playbooks/Process_Email_-_Generic_v2.yml (hunk 2 above)
Comment (anas-yousef) [reply Packs/Phishing/Playbooks/Process_Email_-_Generic_v2.yml:2794]: @idovandijk and I checked the playbook, and we saw that it ran without any problems.

**Review by JasBeilin — APPROVED** [review]

**Pull request merged**